src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,510 @@
+        """
+        cmd (byte str): command to be sent to device (without the CR)
+        """
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+            time.sleep(0.05)
+
+    def _sendQuery(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the CR, but with the ?)
+        returns (byte str): answer received from the device (without \n or \r)
+        raise:
+            IOError if no answer is returned in time
+        """
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+
+            self._serial.timeout = 1
+            ans = b''
+            while ans[-1:] != b'\r':
+                char = self._serial.read()
+                if not char:
+                    raise IOError("Timeout after receiving %s" % to_str_escape(ans))
+                ans += char
+
+            logging.debug("Received answer %s", to_str_escape(ans))
+
+            time.sleep(0.05)
+            return ans.strip()
+
+    # Low level serial commands.
+    # Note: These all convert to internal units of the controller
+
+    def GetStatusByte(self):
+        # Checks the device status event register
+        return int(self._sendQuery(b"*ESR?"))
+
+    def ClearStatusByte(self):
+        # Clear the status register after checking it
+        self._sendOrder(b"*CLS")
+
+    def checkError(self):
+        # Checks if an error occurred and raises an exception accordingly.
+        status_byte = self.GetStatusByte()
+        self.ClearStatusByte()
+        
+        errors = []
+        
+        for err in (COMMAND_ERROR, EXECUTION_ERROR, QUERY_ERROR):
+            if status_byte & err:
+                errors.append(STATUS_BYTE[err])
+
+        if errors:
+            raise LakeshoreError("Lakeshore %s" % (str(errors)), status_byte)
+
+    def GetIdentifier(self):
+        """
+        Get the identifier from the controller
+        Returns 3 strings: manufacturer, model number, and serial number
+        """
+        identity = self._sendQuery(b'*IDN?')
+        manufacturer, md, serialn, firmware = identity.decode("latin1").split(',')
+        return manufacturer, md, serialn, firmware
+    
+    def GetTemp(self):
+        """
+        Get the temperature at the thermocouple junction
+        """
+        return float(self._sendQuery(b"TEMP?"))
+
+    def SetSetpoint(self, temp):
+        """
+        Set the temperature setpoint
+        temp (float): the temperature to set, in Celsius
+        """
+        self._sendOrder(b"SETP %d,%.2f" % (self._output_channel, temp + KELVIN_CONVERT))
+
+    def GetSetpoint(self):
+        """
+        Get the temperature setpoint. Returns a float in Celsius
+        """
+        val = self._sendQuery(b"SETP? %d" % (self._output_channel,)).strip('+')
```

Comment:
Is the "string(+)" really needed? float() accepts + (if there is no space between the + and the number). So maybe you can drop it to keep the code simpler (and the next function too).

Reply:
Yes, I think so. Somehow float() didn't like it and it was breaking till I added the strip..